src/SprykerEco/Zed/AfterPay/Business/Api/Adapter/Client/Http/Guzzle.php

```diff
@@ -112,28 +112,52 @@ protected function send(RequestInterface $request, array $options = []): Respons
         } catch (RequestException $requestException) {
             $apiHttpRequestException = new ApiHttpRequestException($requestException->getMessage());
 
-            $content = $requestException->getResponse()->getBody()->getContents();
-            if (!empty($content)) {
-                $errorResponseData = $this->encodingService->decodeJson($content, true);
-                if (isset($errorResponseData[0])) {
-                    $errorDetails = $errorResponseData[0];
-
-                    $apiErrorTransfer = (new AfterPayApiResponseErrorTransfer())
-                        ->setActionCode($errorDetails['actionCode'])
-                        ->setCode($errorDetails['code'])
-                        ->setType($errorDetails['type'])
-                        ->setMessage($errorDetails['message'])
-                        ->setIsSuccess(false);
-
-                    $apiHttpRequestException->setError($apiErrorTransfer);
-                    $apiHttpRequestException->setDetailedMessage($content);
-                }
+            $responseContent = $this->getExceptionResponseContent($requestException);
+            $errorsResponseData = $this->encodingService->decodeJson($responseContent, true);
+            $afterPayApiResponseErrorTransfer = $this->createAfterPayApiResponseErrorTransfer($errorsResponseData);
+
+            if (!empty($afterPayApiResponseErrorTransfer)) {
+                $apiHttpRequestException->setError($afterPayApiResponseErrorTransfer);
+                $apiHttpRequestException->setDetailedMessage($responseContent);
             }
 
             throw $apiHttpRequestException;
         }
     }
 
+    /**
+     * @param array|null $errorsResponseData
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\AbstractTransfer\AfterPayApiResponseErrorTransfer|null
+     */
+    protected function createAfterPayApiResponseErrorTransfer(?array $errorsResponseData): ?AfterPayApiResponseErrorTransfer
+    {
+        if (empty($errorsResponseData[0])) {
+            return null;
+        }
+
+        $errorDetails = $errorsResponseData[0];
+
+        return (new AfterPayApiResponseErrorTransfer())
+            ->setActionCode($errorDetails['actionCode'])
```

Comment:
Extract keys to consts, for example:
 * `ERROR_KEY_ACTION_CODE`.
 * `ERROR_KEY_CODE`.
 * `ERROR_KEY_TYPE`.
 * `ERROR_KEY_MESSAGE`.